examples/with_winit/src/stats.rs

```diff
@@ -0,0 +1,169 @@
+//
+// Also licensed under MIT license, at your choice.
+
+use scenes::SimpleText;
+use std::{collections::VecDeque, time::Instant};
+use vello::{
+    kurbo::{Affine, Rect},
+    peniko::{Brush, Color, Fill},
+    SceneBuilder,
+};
+
+const SLIDING_WINDOW_SIZE: usize = 100;
+
+#[derive(Debug)]
+pub struct Snapshot {
+    pub fps: f64,
+    pub frame_time_ms: f64,
+    pub frame_time_min_ms: f64,
+    pub frame_time_max_ms: f64,
+}
+
+impl Snapshot {
+    pub fn draw_layer(
+        &self,
+        sb: &mut SceneBuilder,
+        text: &mut SimpleText,
+        viewport_width: f64,
+        viewport_height: f64,
+    ) {
+        let width = (viewport_width * 0.4).max(200.).min(400.);
+        let height = width * 0.3;
+        let x_offset = viewport_width - width;
+        let y_offset = viewport_height - height;
+        let offset = Affine::translate((x_offset, y_offset));
+        let text_height = height * 0.2;
+        let left_margin = width * 0.03;
+        let text_size = (text_height * 0.9) as f32;
+        sb.fill(
+            Fill::NonZero,
+            offset,
+            &Brush::Solid(Color::rgba8(0, 0, 0, 200)),
+            None,
+            &Rect::new(0., 0., width, height),
+        );
+        text.add(
+            sb,
+            None,
+            text_size,
+            Some(&Brush::Solid(Color::WHITE)),
+            offset * Affine::translate((left_margin, text_height)),
+            &format!("Frame Time: {:.2} ms", self.frame_time_ms),
+        );
+        text.add(
+            sb,
+            None,
+            text_size,
+            Some(&Brush::Solid(Color::WHITE)),
+            offset * Affine::translate((left_margin, 2. * text_height)),
+            &format!("Frame Time (min): {:.2} ms", self.frame_time_min_ms),
+        );
+        text.add(
+            sb,
+            None,
+            text_size,
+            Some(&Brush::Solid(Color::WHITE)),
+            offset * Affine::translate((left_margin, 3. * text_height)),
+            &format!("Frame Time (max): {:.2} ms", self.frame_time_max_ms),
+        );
+        text.add(
+            sb,
+            None,
+            text_size,
+            Some(&Brush::Solid(Color::WHITE)),
+            offset * Affine::translate((left_margin, 4. * text_height)),
+            &format!("Resolution: {viewport_width}x{viewport_height}"),
+        );
+        text.add(
+            sb,
+            None,
+            text_size,
+            Some(&Brush::Solid(Color::WHITE)),
+            offset * Affine::translate((width * 0.67, text_height)),
+            &format!("FPS: {:.2}", self.fps),
+        );
+    }
+}
+
+pub struct Stats {
+    count: usize,
+    sum: u64,
+    min: u64,
+    max: u64,
+    samples: VecDeque<u64>,
+}
+
+pub struct FrameScope<'a> {
+    stats: &'a mut Stats,
+    start: Instant,
+}
+
+impl<'a> Drop for FrameScope<'a> {
+    fn drop(&mut self) {
+        self.stats
+            .add_sample(self.start.elapsed().as_micros() as u64);
```

Comment:
To ensure that this is measuring time between the same point, rather than rendering time, I think we need to inline the call to `elapsed`, store the new `Instant`, and use that as the start time in `frame_scope`. At the moment, we ignore input processing time, for example, which I don't think is right. 

Actually working out the best measure of FPS has always eluded me.

We also should measure GPU time (e.g. using https://github.com/Wumpf/wgpu-profiler), but wiring that up is non-trivial.

Reply:
> We also should measure GPU time (e.g. using https://github.com/Wumpf/wgpu-profiler), but wiring that up is non-trivial.

Oh, the GPU time measurement looks neat. I haven't looked into `wgpu-profiler` very closely but supporting GPU timers out of the box would be great (so far I've had to rely on external tools, such as the Metal GPU Performance monitor). I'll investigate this.

Reply:
I made the elapsed time measurement tighter so it should closely match the presentation time. I haven't really had time to investigate wgpu-profiler yet but I plan to spend some time on wiring it up soon.